doc/api/assert.md

```diff
@@ -45,12 +45,12 @@ Only [enumerable "own" properties][] are considered. The
 non-enumerable properties — for such checks, consider using
 [`assert.deepStrictEqual()`][] instead. This can lead to some
 potentially surprising results. For example, the following example does not
-throw an `AssertionError` because the properties on the [`Error`][] object are
+throw an `AssertionError` because the properties on the [`RegExp`][] object are
 not enumerable:
 
 ```js
 // WARNING: This does not throw an AssertionError!
-assert.deepEqual(Error('a'), Error('b'));
+assert.deepEqual(/a/, {});
```

Comment:
Oh , yeah, I think it's definitely less obvious with a `Date` object that there aren't enumerable properties!

I think something like this might be even better:

```js
assert.deepEqual(new Date(), /a/);
```

My thinking is that those are two objects where the absence of enumerable properties is easy to forget. So that result is genuinely surprising at a glance.

In contrast, the idea that an empty object or an empty array don't have enumerable properties is more obvious.

I might be overthinking this a bit. :-D